server/service/src/sync/test/integration/errors.rs

```diff
@@ -119,29 +118,80 @@ mod tests {
     }
 
     #[actix_rt::test]
-    async fn integration_sync_not_parsed_error() {
-        let central_config = ConfigureCentralServer::from_env();
+    async fn api_incompatible_error() {
+        let SiteConfiguration { sync_settings, .. } = ConfigureCentralServer::from_env()
+            .create_sync_site(vec![])
+            .await
+            .expect("Problem creating sync site");
 
-        // Should result in an error in non standard format
-        let error = match central_config
-            .upsert_records(json!({
-                "this_one_does_not_exist": [{"should_error":  true}]
-            }))
+        let ServiceTestContext {
+            service_provider,
+            service_context,
+            ..
+        } = setup_all_and_service_provider("api_incompatible_error", MockDataInserts::none()).await;
+
+        service_provider
+            .site_info_service
+            .request_and_set_site_info(&service_provider, &sync_settings)
             .await
-        {
-            Ok(_) => panic!("Should result in error"),
-            Err(e) => e,
-        };
+            .unwrap();
+        service_provider
+            .settings
+            .update_sync_settings(&service_context, &sync_settings)
+            .unwrap();
+
+        let synchroniser =
+            Synchroniser::new_with_version(sync_settings.clone(), service_provider, 10000).unwrap();
+
+        let error = synchroniser
+            .sync()
+            .await
+            .err()
+            .expect("Should result in error");
 
         assert_matches!(
             error,
-            SyncApiError {
-                source: SyncApiErrorVariant::AsText {
-                    status: StatusCode::INTERNAL_SERVER_ERROR,
-                    ..
+            SyncError::PostInitialisationError(PostInitialisationError(SyncApiError {
+                source: SyncApiErrorVariant::ParsedError {
+                    status: StatusCode::CONFLICT,
+                    source: ParsedError {
+                        code: SyncErrorCodeV5::ApiVersionIncompatible,
+                        data: Some(_),
+                        ..
+                    }
                 },
                 ..
-            }
+            }))
         );
     }
+
+    // This test was checking for `html` type return from 4d server, this seems to be captured now
```

Comment:
this was weird, I didn't quite delete this test yet, not sure if I should or not, since I do like the test for AsText variant (this happens when we fail to deserialize error input, due to it's shape or json validity), but didn't know how to replicate it now without adding new endpoint